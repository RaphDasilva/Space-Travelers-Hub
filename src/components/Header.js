import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../assets/logo.png';

const Header = () => (
  <nav className="navBar">
    <div className="Logo">
      <img className="top-logo" src={logo} alt="logo" />
      <h1>Space Travelers Hub</h1>
    </div>
    <ul className="links">
      <NavLink className="link" to="/">Rockets</NavLink>
      <NavLink className="link" to="/Missions">Missions</NavLink>
      <div className="vl" />
      <NavLink className="link" to="/Profile">My Profile</NavLink>
    </ul>
  </nav>
);

export default Header;
