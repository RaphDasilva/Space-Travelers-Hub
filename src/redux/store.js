import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './mission/missionSlice';
import rocketsReducer from './rockets/rockets';

const store = configureStore({
  reducer: {
    mission: missionReducer, rocketsReducer,
  },
});

export default store;
