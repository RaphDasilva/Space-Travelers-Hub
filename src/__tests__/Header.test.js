import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Header from '../components/Header';
import store from '../redux/store';

describe('Header component', () => {
  it('should render Header component', () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
