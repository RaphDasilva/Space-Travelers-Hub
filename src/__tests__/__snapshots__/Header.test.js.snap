// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component 1`] = `
<div>
  <nav
    class="navBar"
  >
    <div
      class="Logo"
    >
      <img
        alt="logo"
        class="top-logo"
        src="logo.png"
      />
      <h1>
        Space Travelers Hub
      </h1>
    </div>
    <ul
      class="links"
    >
      <a
        aria-current="page"
        class="link active"
        href="/"
      >
        Rockets
      </a>
      <a
        class="link"
        href="/Missions"
      >
        Missions
      </a>
      <div
        class="vl"
      />
      <a
        class="link"
        href="/Profile"
      >
        My Profile
      </a>
    </ul>
  </nav>
</div>
`;
